### PR TITLE
Allow sanitize_all and its CLI to run on multiple cores

### PR DESCRIPTION
**Pull request by wizmer**

### CODE REVIEW

File: neuror/sanitize.py [hunk 1]
```diff
@@ -41,7 +42,28 @@ def sanitize(input_neuron, output_path):
...
 
-def sanitize_all(input_folder, output_folder):
+def _sanitize_one(args):
+    '''Function to be called by sanitize_all to catch all exceptions
+    and return path if in error
+
+    Since Pool.imap_unordered only supports one argument, the argument
+    is a tuple: (path, input_folder, output_folder).
+    '''
+    path, input_folder, output_folder = args
```
Comment (arnaudon) [inline neuror/sanitize.py:52]: I would use partial to fix the two input/output argument as they are constants

File: neuror/sanitize.py [hunk 2]
```diff
@@ -56,16 +78,13 @@ def sanitize_all(input_folder, output_folder):
...
-        relative_path = path.relative_to(input_folder)
-        output_dir = output_folder / relative_path.parent
-        if not output_dir.exists():
-            output_dir.mkdir(parents=True)
-        try:
-            sanitize(path, output_dir / path.name)
-        except (MorphioError, CorruptedMorphology):
-            errored_paths.append(str(path))
+    morphologies = list(iter_morphologies(Path(input_folder)))
+    args = ((morphology, input_folder, output_folder) for morphology in morphologies)
```
Comment (arnaudon) [inline neuror/sanitize.py:82]: here you can give the iter_morphologies to pool.imap directly

File: neuror/sanitize.py [hunk 3]
```diff
@@ -56,16 +78,13 @@ def sanitize_all(input_folder, output_folder):
...
+    if nprocesses == 1:
+        results = map(_sanitize_one, args)
+    else:
+        results = Pool(nprocesses).imap_unordered(_sanitize_one, args, chunksize=100)
```
Comment (arnaudon) [inline neuror/sanitize.py:86]: I think chunksize here is not needed, as the function is small (nothing heavy to put in memory, etc...)

File: neuror/sanitize.py [hunk 4]
```diff
@@ -56,16 +78,13 @@ def sanitize_all(input_folder, output_folder):
...
+    errored_paths = list(filter(None, tqdm(results, total=len(morphologies))))
```
Comment (arnaudon) [inline neuror/sanitize.py:87]: ah then you can't get the total if you use iterator only above, maybe just use list(iter_morphologies(

File: neuror/sanitize.py (hunk 1 above)
Comment (wizmer) [reply neuror/sanitize.py:52]: Good point

File: neuror/sanitize.py (hunk 2 above)
Comment (wizmer) [reply neuror/sanitize.py:82]: Yes, as you said in the next comment, I create a list to know the number of items for `tqdm`

File: neuror/sanitize.py [hunk 5]
```diff
@@ -41,7 +43,27 @@ def sanitize(input_neuron, output_path):
     fix_non_zero_segments(neuron).write(str(output_path))
 
 
-def sanitize_all(input_folder, output_folder):
+def _sanitize_one(path, input_folder, output_folder):
+    '''Function to be called by sanitize_all to catch all exceptions
+    and return path if in error
+
+    Since Pool.imap_unordered only supports one argument, the argument
```
Comment (asanin-epfl) [inline neuror/sanitize.py:50]: This comment can also be removed, I guess.

Comment (asanin-epfl) [reply neuror/sanitize.py:50]: @wizmer what about this comment?

**Pull request merged**